platform/osgi/platform-osgi-configadmin/src/main/java/org/codice/felix/cm/file/EncryptingPersistenceManager.java

```diff
@@ -53,25 +57,39 @@
 
   private static final Set<String> EXCLUDED_PROPERTIES = new HashSet<>();
 
-  // Not worth adding a dependency on felix's fileinstall module just for the constant
-  static final String FELIX_FILENAME = "felix.fileinstall.filename";
-
   static {
     EXCLUDED_PROPERTIES.add(SERVICE_PID);
     EXCLUDED_PROPERTIES.add(SERVICE_FACTORYPID);
     EXCLUDED_PROPERTIES.add(FELIX_FILENAME);
   }
 
+  private class InitOperation implements PrivilegedAction<EncryptionAgent> {
+    private final String passwordDirectory;
+
+    InitOperation(String passwordDirectory) {
+      this.passwordDirectory = passwordDirectory;
+    }
+
+    @Override
+    public EncryptionAgent run() {
+      return new EncryptionAgent(passwordDirectory);
+    }
+  }
+
   private final EncryptionAgent agent;
 
   public EncryptingPersistenceManager(PersistenceManager persistenceManager) {
-    this(persistenceManager, System.getProperty("ddf.home").concat("/etc/certs/bundle-cache"));
+    this(
+        persistenceManager,
+        FrameworkUtil.getBundle(EncryptingPersistenceManager.class)
+            .getDataFile("")
+            .getAbsolutePath());
   }
 
   public EncryptingPersistenceManager(
       PersistenceManager persistenceManager, String passwordDirectory) {
     super(persistenceManager);
-    this.agent = new EncryptionAgent(passwordDirectory);
+    this.agent = AccessController.doPrivileged(new InitOperation(passwordDirectory));
```

Comment:
❗️ I would change this to an anonymous class as opposed to creating a named class. This is how it was meant to be used. Would definitely be nice if they had made that interface a functional one

Reply:
The problem is that they overloaded the `doPrivileged` method. You can pass a function, but you still have to cast it so the compiler can tell if it's a `PrivilegedAction` or a `PrivilegedExceptionAction`.

But definitely do this anon/with a cast function.

```java
    this.agent = AccessController.doPrivileged(
        (PrivilegedAction<EncryptionAgent>) () -> new EncryptionAgent((passwordDirectory)));
```
(`passwordDirectory` is effectively final)

Reply:
❗️  Cannot use Java 8 constructs in the config admin bundle - see above comment:
https://github.com/codice/ddf/pull/2523#discussion_r147325495

Reply:
😞  Add a comment to the file to that regard.

Reply:
❗️ It appears that comments have not yet been added to the java files that cannot use Java 8 constructs.

Reply:
☝️